<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>8. Third Party Tools &mdash; EDGE 1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="shortcut icon" href="_static/edge.ico"/>
    <link rel="copyright" title="Copyright" href="copyright.html" />
    <link rel="top" title="EDGE 1.0 documentation" href="index.html" />
    <link rel="next" title="9. Troubleshooting" href="troubleshooting.html" />
    <link rel="prev" title="7. Databases" href="database.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="troubleshooting.html" title="9. Troubleshooting"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="database.html" title="7. Databases"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">EDGE 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="third-party-tools">
<span id="id1"></span><h1>8. Third Party Tools<a class="headerlink" href="#third-party-tools" title="Permalink to this headline">¶</a></h1>
<div class="section" id="assembly">
<h2>8.1. Assembly<a class="headerlink" href="#assembly" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>IDBA-UD<ul>
<li>Citation: Peng, Y., et al. (2012) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/22495754">IDBA-UD: a de novo assembler for single-cell and metagenomic sequencing data with highly uneven depth, Bioinformatics, 28, 1420-1428.</a></li>
<li>Site: <a class="reference external" href="http://i.cs.hku.hk/~alse/hkubrg/projects/idba_ud/">http://i.cs.hku.hk/~alse/hkubrg/projects/idba_ud/</a></li>
<li>Version: 1.1.1</li>
<li>License: GPLv2</li>
</ul>
</li>
<li>RATT<ul>
<li>Citation: Otto, T.D., et al. (2011) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/21306991">RATT: Rapid Annotation Transfer Tool, Nucleic acids research, 39, e57.</a></li>
<li>Site: <a class="reference external" href="http://ratt.sourceforge.net/">http://ratt.sourceforge.net/</a></li>
<li>Version:</li>
<li>License:</li>
</ul>
</li>
</ul>
</div>
<div class="section" id="annotation">
<h2>8.2. Annotation<a class="headerlink" href="#annotation" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>Prokka<ul>
<li>Citation: Seemann, T. (2014) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/24642063">Prokka: rapid prokaryotic genome annotation, Bioinformatics, 30,2068-2069.</a></li>
<li>Site: <a class="reference external" href="http://www.vicbioinformatics.com/software.prokka.shtml">http://www.vicbioinformatics.com/software.prokka.shtml</a></li>
<li>Version: 1.9</li>
<li>License: GPLv2</li>
</ul>
</li>
<li>tRNAscan<ul>
<li>Citation: Lowe, T.M. and Eddy, S.R. (1997) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/9023104">tRNAscan-SE: a program for improved detection of transfer RNA genes in genomic sequence, Nucleic acids research, 25, 955-964.</a></li>
<li>Site: <a class="reference external" href="http://lowelab.ucsc.edu/tRNAscan-SE/">http://lowelab.ucsc.edu/tRNAscan-SE/</a></li>
<li>Version: 1.3.1</li>
<li>License: GPLv2</li>
</ul>
</li>
<li>Barrnap<ul>
<li>Citation:</li>
<li>Site: <a class="reference external" href="http://www.vicbioinformatics.com/software.barrnap.shtml">http://www.vicbioinformatics.com/software.barrnap.shtml</a></li>
<li>Version: 0.42</li>
<li>License: GPLv3</li>
</ul>
</li>
<li>BLAST+<ul>
<li>Citation: Camacho, C., et al. (2009) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/20003500">BLAST+: architecture and applications, BMC bioinformatics, 10, 421.</a></li>
<li>Site: <a class="reference external" href="ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/2.2.29/">ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/2.2.29/</a></li>
<li>Version: 2.2.29</li>
<li>License: Public domain</li>
</ul>
</li>
<li>blastall<ul>
<li>Citation: Altschul, S.F., et al. (1990) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/2231712">Basic local alignment search tool, Journal of molecular biology, 215, 403-410.</a></li>
<li>Site: <a class="reference external" href="ftp://ftp.ncbi.nlm.nih.gov/blast/executables/release/2.2.26/">ftp://ftp.ncbi.nlm.nih.gov/blast/executables/release/2.2.26/</a></li>
<li>Version: 2.2.26</li>
<li>License: Public domain</li>
</ul>
</li>
<li>Phage_Finder<ul>
<li>Citation: Fouts, D.E. (2006) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/17062630">Phage_Finder: automated identification and classification of prophage regions in complete bacterial genome sequences, Nucleic acids research, 34, 5839-5851.</a></li>
<li>Site: <a class="reference external" href="http://phage-finder.sourceforge.net/">http://phage-finder.sourceforge.net/</a></li>
<li>Version: 2.1</li>
<li>License: GPLv3</li>
</ul>
</li>
<li>Glimmer<ul>
<li>Citation: Delcher, A.L., et al. (2007) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/17237039">Identifying bacterial genes and endosymbiont DNA with Glimmer, Bioinformatics, 23, 673-679.</a></li>
<li>Site: <a class="reference external" href="http://ccb.jhu.edu/software/glimmer/index.shtml">http://ccb.jhu.edu/software/glimmer/index.shtml</a></li>
<li>Version: 302b</li>
<li>License: Artistic License</li>
</ul>
</li>
<li>ARAGORN<ul>
<li>Citation: Laslett, D. and Canback, B. (2004) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/14704338">ARAGORN, a program to detect tRNA genes and tmRNA genes in nucleotide sequences, Nucleic acids research, 32, 11-16.</a></li>
<li>Site: <a class="reference external" href="http://mbio-serv2.mbioekol.lu.se/ARAGORN/">http://mbio-serv2.mbioekol.lu.se/ARAGORN/</a></li>
<li>Version: 1.2.36</li>
<li>License:</li>
</ul>
</li>
<li>Prodigal<ul>
<li>Citation: Hyatt, D., et al. (2010) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/20211023">Prodigal: prokaryotic gene recognition and translation initiation site identification, BMC bioinformatics, 11, 119.</a></li>
<li>Site: <a class="reference external" href="http://prodigal.ornl.gov/">http://prodigal.ornl.gov/</a></li>
<li>Version: 2_60</li>
<li>License: GPLv3</li>
</ul>
</li>
<li>tbl2asn<ul>
<li>Citation:</li>
<li>Site: <a class="reference external" href="http://www.ncbi.nlm.nih.gov/genbank/tbl2asn2/">http://www.ncbi.nlm.nih.gov/genbank/tbl2asn2/</a></li>
<li>Version: 23.6 (2014 Oct 20th)</li>
<li>License:</li>
</ul>
</li>
</ul>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">tbl2asn will expire in a year. Need to update it every year.</p>
</div>
</div>
<div class="section" id="alignment">
<h2>8.3. Alignment<a class="headerlink" href="#alignment" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>HMMER3<ul>
<li>Citation: Eddy, S.R. (2011) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/22039361">Accelerated Profile HMM Searches, PLoS computational biology, 7, e1002195</a></li>
<li>Site: <a class="reference external" href="http://hmmer.janelia.org/">http://hmmer.janelia.org/</a></li>
<li>Version: 3.1b1</li>
<li>License: GPLv3</li>
</ul>
</li>
<li>Infernal<ul>
<li>Citation: Nawrocki, E.P. and Eddy, S.R. (2013) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/24008419">Infernal 1.1: 100-fold faster RNA homology searches, Bioinformatics, 29, 2933-2935.</a></li>
<li>Site: <a class="reference external" href="http://infernal.janelia.org/">http://infernal.janelia.org/</a></li>
<li>Version: 1.1rc4</li>
<li>License: GPLv3</li>
</ul>
</li>
<li>Bowtie 2<ul>
<li>Citation: Langmead, B. and Salzberg, S.L. (2012) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/22388286">Fast gapped-read alignment with Bowtie 2, Nature methods, 9, 357-359.</a></li>
<li>Site: <a class="reference external" href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">http://bowtie-bio.sourceforge.net/bowtie2/index.shtml</a></li>
<li>Version: 2.1.0</li>
<li>License: GPLv3</li>
</ul>
</li>
<li>BWA<ul>
<li>Citation: Li, H. and Durbin, R. (2009) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/19451168">Fast and accurate short read alignment with Burrows-Wheeler transform, Bioinformatics, 25, 1754-1760.</a></li>
<li>Site: <a class="reference external" href="http://bio-bwa.sourceforge.net/">http://bio-bwa.sourceforge.net/</a></li>
<li>Version: 0.7.9</li>
<li>License: GPLv3</li>
</ul>
</li>
<li>MUMmer3<ul>
<li>Citation: Kurtz, S., et al. (2004) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/14759262">Versatile and open software for comparing large genomes, Genome biology, 5, R12.</a></li>
<li>Site: <a class="reference external" href="http://mummer.sourceforge.net/">http://mummer.sourceforge.net/</a></li>
<li>Version: 3.23</li>
<li>License: GPLv3</li>
</ul>
</li>
</ul>
</div>
<div class="section" id="taxonomy-classification">
<h2>8.4. Taxonomy Classification<a class="headerlink" href="#taxonomy-classification" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>Kraken<ul>
<li>Citation: Wood, D.E. and Salzberg, S.L. (2014) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/24580807">Kraken: ultrafast metagenomic sequence classification using exact alignments, Genome biology, 15, R46.</a></li>
<li>Site: <a class="reference external" href="http://ccb.jhu.edu/software/kraken/">http://ccb.jhu.edu/software/kraken/</a></li>
<li>Version: 0.10.4-beta</li>
<li>License: GPLv3</li>
</ul>
</li>
<li>Metaphlan<ul>
<li>Citation: Segata, N., et al. (2012) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/22688413">Metagenomic microbial community profiling using unique clade-specific marker genes, Nature methods, 9, 811-814.</a></li>
<li>Site: <a class="reference external" href="http://huttenhower.sph.harvard.edu/metaphlan">http://huttenhower.sph.harvard.edu/metaphlan</a></li>
<li>Version: 1.7.7</li>
<li>License: Artistic License</li>
</ul>
</li>
<li>MetaPhylerSR<ul>
<li>Citation: Liu, B., et al. (2011) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/21989143">Accurate and fast estimation of taxonomic profiles from metagenomic shotgun sequences, BMC genomics, 12 Suppl 2, S4.</a></li>
<li>Site: <a class="reference external" href="http://metaphyler.cbcb.umd.edu/">http://metaphyler.cbcb.umd.edu/</a></li>
<li>Version: 0.115</li>
<li>License:</li>
</ul>
</li>
<li>GOTTCHA<ul>
<li>Citation: Tracey Allen K. Freitas, Po-E Li, Matthew B. Scholz, Patrick S. G. Chain: Accurate Metagenome characterization using a hierarchical suite of unique signatures. (in submission)</li>
<li>Site: <a class="reference external" href="https://github.com/LANL-Bioinformatics/GOTTCHA">https://github.com/LANL-Bioinformatics/GOTTCHA</a></li>
<li>Version: 0.9d</li>
<li>License: GPLv3</li>
</ul>
</li>
<li>metascope_plus<ul>
<li>Citation:</li>
<li>Site:</li>
<li>Version:</li>
<li>License:</li>
</ul>
</li>
</ul>
</div>
<div class="section" id="phylogeny">
<h2>8.5. Phylogeny<a class="headerlink" href="#phylogeny" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>FastTree<ul>
<li>Citation: Morgan N. Price, Paramvir S. Dehal, and Adam P. Arkin. 2009. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/19377059">FastTree: Computing Large Minimum Evolution Trees with Profiles instead of a Distance Matrix. Mol Biol Evol (2009) 26 (7): 1641-1650</a></li>
<li>Site: <a class="reference external" href="http://www.microbesonline.org/fasttree/">http://www.microbesonline.org/fasttree/</a></li>
<li>Version: 2.1.7</li>
<li>License: GPLv2</li>
</ul>
</li>
<li>RAxML<ul>
<li>Citation: Stamatakis,A. 2014. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/24451623">RAxML version 8: A tool for phylogenetic analysis and post-analysis of large phylogenies. Bioinformatics, 30:1312-1313</a></li>
<li>Site: <a class="reference external" href="http://sco.h-its.org/exelixis/web/software/raxml/index.html">http://sco.h-its.org/exelixis/web/software/raxml/index.html</a></li>
<li>Version: 8.0.26</li>
<li>License: GPLv2</li>
</ul>
</li>
<li>Bio::Phylo<ul>
<li>Citation: Rutger A Vos, Jason Caravas, Klaas Hartmann, Mark A Jensen and Chase Miller, (2011). <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/21352572">Bio::Phylo - phyloinformatic analysis using Perl. BMC Bioinformatics 12:63.</a></li>
<li>Site: <a class="reference external" href="http://search.cpan.org/~rvosa/Bio-Phylo/">http://search.cpan.org/~rvosa/Bio-Phylo/</a></li>
<li>Version: 0.58</li>
<li>License: GPLv3</li>
</ul>
</li>
</ul>
</div>
<div class="section" id="visualization-and-graphic-user-interface">
<h2>8.6. Visualization and Graphic User Interface<a class="headerlink" href="#visualization-and-graphic-user-interface" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>JQuery Mobile<ul>
<li>Site: <a class="reference external" href="http://jquerymobile.com">http://jquerymobile.com</a></li>
<li>Version :1.4.3</li>
<li>License: CC0</li>
</ul>
</li>
<li>jsPhyloSVG<ul>
<li>Citation: Smits SA, Ouverney CC, (2010) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/20805892">jsPhyloSVG: A Javascript Library for Visualizing Interactive and Vector-Based Phylogenetic Trees on the Web. PLoS ONE 5(8): e12267.</a></li>
<li>Site: <a class="reference external" href="http://www.jsphylosvg.com">http://www.jsphylosvg.com</a></li>
<li>Version: 1.55</li>
<li>License: GPL</li>
</ul>
</li>
<li>JBrowse<ul>
<li>Citation: Skinner, M.E., et al. (2009) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/19570905">JBrowse: a next-generation genome browser, Genome research, 19, 1630-1638.</a></li>
<li>Site: <a class="reference external" href="http://jbrowse.org">http://jbrowse.org</a></li>
<li>Version: 1.11.5</li>
<li>License: Artistic License 2.0/LGPLv.1</li>
</ul>
</li>
<li>KronaTools<ul>
<li>Citation: Ondov, B.D., Bergman, N.H. and Phillippy, A.M. (2011) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/21961884">Interactive metagenomic visualization in a Web browser, BMC bioinformatics, 12, 385.</a></li>
<li>Site: <a class="reference external" href="http://sourceforge.net/projects/krona/">http://sourceforge.net/projects/krona/</a></li>
<li>Version: 2.4</li>
<li>License: BSD</li>
</ul>
</li>
</ul>
</div>
<div class="section" id="utility">
<h2>8.7. Utility<a class="headerlink" href="#utility" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>BEDTools<ul>
<li>Citation: Quinlan, A.R. and Hall, I.M. (2010) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/20110278">BEDTools: a flexible suite of utilities for comparing genomic features, Bioinformatics, 26, 841-842.</a></li>
<li>Site: <a class="reference external" href="https://github.com/arq5x/bedtools2">https://github.com/arq5x/bedtools2</a></li>
<li>Version: 2.19.1</li>
<li>License: GPLv2</li>
</ul>
</li>
<li>R<ul>
<li>Citation: R Core Team (2013). R: A language and environment for statistical   computing. R Foundation for Statistical Computing, Vienna, Austria.   URL <a class="reference external" href="http://www.R-project.org/">http://www.R-project.org/</a>.</li>
<li>Site: <a class="reference external" href="http://www.r-project.org/">http://www.r-project.org/</a></li>
<li>Version: 2.15.3</li>
<li>License: GPLv2</li>
</ul>
</li>
<li>GNU_parallel<ul>
<li>Citation: O. Tange (2011): GNU Parallel - The Command-Line Power Tool, ;login: The USENIX Magazine, February 2011:42-47</li>
<li>Site: <a class="reference external" href="http://www.gnu.org/software/parallel/">http://www.gnu.org/software/parallel/</a></li>
<li>Version: 20140622</li>
<li>License: GPLv3</li>
</ul>
</li>
<li>tabix<ul>
<li>Citation:</li>
<li>Site: <a class="reference external" href="http://sourceforge.net/projects/samtools/files/tabix/">http://sourceforge.net/projects/samtools/files/tabix/</a></li>
<li>Version: 0.2.6</li>
<li>License:</li>
</ul>
</li>
<li>Primer3<ul>
<li>Citation: Untergasser, A., et al. (2012) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/22730293">Primer3&#8211;new capabilities and interfaces, Nucleic acids research, 40, e115.</a></li>
<li>Site: <a class="reference external" href="http://primer3.sourceforge.net/">http://primer3.sourceforge.net/</a></li>
<li>Version: 2.3.5</li>
<li>License: GPLv2</li>
</ul>
</li>
<li>SAMtools<ul>
<li>Citation: Li, H., et al. (2009) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/19505943">The Sequence Alignment/Map format and SAMtools, Bioinformatics, 25, 2078-2079.</a></li>
<li>Site: <a class="reference external" href="http://samtools.sourceforge.net/">http://samtools.sourceforge.net/</a></li>
<li>Version: 0.1.19</li>
<li>License: MIT</li>
</ul>
</li>
</ul>
<ul class="simple" id="faqcs-ref">
<li>FaQCs<ul>
<li>Citation: Chienchi Lo, PatrickS.G. Chain (2014) Rapid evaluation and Quality Control of Next Generation Sequencing Data with FaQCs. (in print)</li>
<li>Site: <a class="reference external" href="https://github.com/LANL-Bioinformatics/FaQCs">https://github.com/LANL-Bioinformatics/FaQCs</a></li>
<li>Version: 1.33</li>
<li>License: GPLv3</li>
</ul>
</li>
<li>wigToBigWig<ul>
<li>Citation: Kent, W.J., et al. (2010) <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/20639541">BigWig and BigBed: enabling browsing of large distributed datasets, Bioinformatics, 26, 2204-2207.</a></li>
<li>Site: <a class="reference external" href="https://genome.ucsc.edu/goldenPath/help/bigWig.html#Ex3">https://genome.ucsc.edu/goldenPath/help/bigWig.html#Ex3</a></li>
<li>Version: 4</li>
<li>License:</li>
</ul>
</li>
</ul>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">8. Third Party Tools</a><ul>
<li><a class="reference internal" href="#assembly">8.1. Assembly</a></li>
<li><a class="reference internal" href="#annotation">8.2. Annotation</a></li>
<li><a class="reference internal" href="#alignment">8.3. Alignment</a></li>
<li><a class="reference internal" href="#taxonomy-classification">8.4. Taxonomy Classification</a></li>
<li><a class="reference internal" href="#phylogeny">8.5. Phylogeny</a></li>
<li><a class="reference internal" href="#visualization-and-graphic-user-interface">8.6. Visualization and Graphic User Interface</a></li>
<li><a class="reference internal" href="#utility">8.7. Utility</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="database.html"
                        title="previous chapter">7. Databases</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="troubleshooting.html"
                        title="next chapter">9. Troubleshooting</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/third_party.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="troubleshooting.html" title="9. Troubleshooting"
             >next</a> |</li>
        <li class="right" >
          <a href="database.html" title="7. Databases"
             >previous</a> |</li>
        <li><a href="index.html">EDGE 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; <a href="copyright.html">Copyright</a> 2014, EDGE development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2.3.
    </div>
  </body>
</html>